docker/prod/wait_for_db.rb

```diff
@@ -0,0 +1,19 @@
+require 'timeout'
+
+timeout = (ENV['WAIT_FOR_DB_TIMEOUT_SECONDS'] || 120).to_i
+
+Timeout::timeout(timeout) do
+  loop do
+    puts "[#{DateTime.now}] waiting for db"
+    sleep 4
+
+    begin
+      ActiveRecord::Migration.check_pending!
```

Comment:
As `check_pending!` is raising a `ActiveRecord::PendingMigrationError` when the db is not migrated yet, this can loop until the timeout is elapsed, which may not be the expected behavior (assuming the behavior is just to wait for db to be up and listening for connection).

I'm not sure where the script `docker/prod/wait-for-db` is used. I suppose it's used in parallel of starting some web and worker containers while an init container is migrating the db, in this case the above behavior is ok, but it could be more obvious if the message was `waiting for db to be up and migrated`.

Reply:
Hi @cbliard , the script is used in an init container in the helm chart: https://github.com/opf/helm-charts/pull/22/files

After the timeout, the k8s deployment will be set to failed and can be restarted (or the timeout increased). What we could do is check for an existing lock on the database migrations (seeing if there's an active migration process).

For now, I've adapted the output as you suggested